### DISCUSSION TITLE
Secondary development

### DISCUSSION THREAD

**Discussion by Ivan-kitty** [dbeaver/dbeaver#11252] [DBeaver Development]
I want to do secondary development for dbeaver, but when I am in eclipse build, report all kinds of errors. Is there any API document for secondary development

**Comment by LonwoLonwo** [comment]
As a first step I can recommend https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse
Eclipse version is also important. You can use last stable version (example 2020-12). Often placing Eclipse in Program Files also leads to errors.
